.github/workflows/make_api_responses.yml

```diff
@@ -0,0 +1,141 @@
+        run: |
+          mkdir test_node1 test_node2
+    
+      # Test node 1 will be a full_stack node, providing the federation services
+      - name: Get recipes for Test node 1
+        uses: actions/checkout@v4
+        with:
+          repository: neurobagel/recipes
+          path: test_node1/recipes
+
+      - name: Get data source for Test node 1
+        uses: actions/checkout@v4
+        with:
+          repository: neurobagel/openneuro-annotations
+          path: test_node1/openneuro-annotations
+
+      - name: Get data for Test node 1
+        run: |
+          mkdir test_node1/openneuro_mini_data
+          cp test_node1/openneuro-annotations/ds000001.jsonld test_node1/openneuro-annotations/ds000002.jsonld test_node1/openneuro_mini_data
+
+      - name: Configure Test node 1
+        working-directory: test_node1/recipes
+        run: |
+          cp template.env .env
+
+          # Set environment variables
+          # We need a .* in the sed command to match even if the line is commented out
+          sed -i 's|^#\?[[:space:]]\?LOCAL_GRAPH_DATA=.*|LOCAL_GRAPH_DATA=../openneuro_mini_data|' .env
+          sed -i 's|^#\?[[:space:]]\?NB_FEDERATE_REMOTE_PUBLIC_NODES=.*|NB_FEDERATE_REMOTE_PUBLIC_NODES=False|' .env
+          sed -i 's|^#\?[[:space:]]\?NB_API_QUERY_URL=.*|NB_API_QUERY_URL=http://localhost:8080|' .env
+
+          # Use full container names to avoid conflicts
+          echo '[{"NodeName": "2 OpenNeuro Datasets", "ApiURL": "http://neurobagel_node-api-1:8000"},{"NodeName": "BIDS Synthetic", "ApiURL": "http://neurobagel_node2-api-1:8000"}]' > local_nb_nodes.json
+
+      - name: Launch Test node 1
+        working-directory: test_node1/recipes
+        run: |
+          docker compose up -d
+
+          # Wait for graph to finish being set up
+          until [ -f scripts/logs/DEPLOY.log ] && grep -q "Finished setting up the Neurobagel graph backend." scripts/logs/DEPLOY.log; do
+            sleep 1
+          done
+
+      - name: Generate partial success result
+        run: |
+          curl -s http://localhost:8080/query | jq . > api-responses/fapi_query_partial_success_207.json
+
+      - name: Generate a fail result
+        run: |
+          docker stop neurobagel_node-api-1
+          curl -s http://localhost:8080/query | jq . > api-responses/fapi_query_fail_207.json
+
+      - name: Get recipes for Test node 2
+        uses: actions/checkout@v4
+        with:
+          repository: neurobagel/recipes
+          path: test_node2/recipes
+
+      - name: Configure Test node 2
+        working-directory: test_node2/recipes
+        run: |
+          cp template.env .env
+
+          # Set environment variables
+          sed -i 's|^#\?[[:space:]]\?COMPOSE_PROJECT_NAME=.*|COMPOSE_PROJECT_NAME=neurobagel_node2|' .env
+          sed -i 's|^#\?[[:space:]]\?NB_GRAPH_PORT_HOST=.*|NB_GRAPH_PORT_HOST=7201|' .env
+          sed -i 's|^#\?[[:space:]]\?NB_NAPI_PORT_HOST=.*|NB_NAPI_PORT_HOST=8001|' .env
```

Comment:
is that something we could reuse across workflows, i.e. with the workflow for the new e2e tests in recipes?

Reply:
I had that thought as well, I think we can revisit/reference this workflow as we update the e2e tests. At present the steps here are a bit more complicated since we need to stop specific containers and send curl requests, and then restart them again, and send more curl requests, so it's not as straightforward to factor out the Docker stack launching.